### PR TITLE
OCLOMRS-321: Implement view all mappings

### PR DESCRIPTION
**Pull request by dessHub**
# JIRA TICKET NAME:
[OCLOMRS-321: Implement view all mappings ](https://issues.openmrs.org/browse/OCLOMRS-321)

# Summary:
- In this ticket, I worked on functionality to display given dictionary mappings. A user can filter mappings by all mapping's attributes e.g `map-type`, `from-concept-name`

### CODE REVIEW

File: src/tests/dictionaryConcepts/container/DictionaryMappings.test.jsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+      mappings: [{
+        to_concept_name: 'random_name',
+        from_concept_name: 'random_name',
+        map_type: 'Map As Type',
+        source: 'random source',
+      }],
+    };
+
+    app.setState({ searchInput: 'ciel' });
+    app.setProps(newProps);
```
Comment (Duncanian) [inline src/tests/dictionaryConcepts/container/DictionaryMappings.test.jsx]: @dessHub , you need to pass an `expect` statement to make the comparison

Comment (dessHub) [reply src/tests/dictionaryConcepts/container/DictionaryMappings.test.jsx]: Fixed it. I didn't actually need this edge case.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 2]
```diff
@@ -72,6 +72,7 @@ export class DictionaryConcepts extends Component {
     localStorage.setItem('type', this.props.match.params.type);
     localStorage.setItem('typeName', this.props.match.params.typeName);
     localStorage.setItem('dictionaryName', this.props.match.params.dictionaryName);
+    localStorage.setItem('params', this.props.match.params);
```
Comment (abulojoshua1) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dessHub, I think it would be best if all props were destructured.

File: src/components/dictionaryConcepts/containers/DictionaryMappings.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    super(props);
+    this.state = {
+      mappingLimit: 10,
+    };
+    autoBind(this);
+  }
+
+  componentDidMount() {
+    const { match: { params: { type } } } = this.props;
+    this.props.fetchDictionaryMappings(type);
```
Comment (abulojoshua1) [inline src/components/dictionaryConcepts/containers/DictionaryMappings.jsx]: @dessHub, I think you should destructure props here.

File: src/components/dictionaryConcepts/containers/DictionaryMappings.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    loading: PropTypes.bool.isRequired,
+    fetchDictionaryMappings: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      mappingLimit: 10,
+    };
+    autoBind(this);
```
Comment (collinewait) [inline src/components/dictionaryConcepts/containers/DictionaryMappings.jsx]: Do we need this `autoBind(this)` when we are using arrow functions?

File: src/styles/dictionary_concepts.scss [hunk 5]
```diff
@@ -6,6 +6,14 @@
   width: auto;
 }
 
+.custom-dictionary-mappings {
+  text-align: left;
+  margin-top: 0%;
```
Comment (abulojoshua1) [inline src/styles/dictionary_concepts.scss]: @dessHub, does `margin-top: 0%` have an effect on the interface styling?

Comment (dessHub) [reply src/styles/dictionary_concepts.scss]: This was for testing purposes, I forgot to remove it. On it.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 2 above)
Comment (dessHub) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @abulojoshua1 I only added one line so I didn't want to destructure since line 72 to 74 wasn't mine. I think it is best if we handle it in an independent ticket.

File: src/components/dictionaryConcepts/containers/DictionaryMappings.jsx (hunk 4 above)
Comment (dessHub) [reply src/components/dictionaryConcepts/containers/DictionaryMappings.jsx]: We actually don't need it @collinewait. I have removed it. Thanks for pointing it out

File: src/components/dictionaryConcepts/containers/DictionaryMappings.jsx (hunk 3 above)
Comment (dessHub) [reply src/components/dictionaryConcepts/containers/DictionaryMappings.jsx]: @abulojoshua1 the reason I didn't include `fetchDictionaryMappings` in destructure object is that `fetchDictionaryMappings` is already declared in the upper scope when I was importing it.

**Comment by dessHub** [comment]
@Duncanian I have fixed it.

**Comment by dkayiwa** [comment]
Closing because we do not need this feature
